python/psautohint/fdTools.py

```diff
@@ -220,14 +217,10 @@ def buildBlueLists(self):
 
     def __repr__(self):
         printStr = []
-        keys = dir(self)
-        for key in keys:
-            val = getattr(self, key)
-            if (val is None) or (
-               isinstance(val, types.MethodType)) or key.startswith("_"):
+        for key, val in vars(self).items():
+            if val is None or key.startswith("_"):
```

Comment:
I don't think there's any key that starts with `_` anymore.

It seems that `fdTools.py` is a trimmed down version of `ConvertFontToCID.py`. I've been working on that file to fix https://github.com/adobe-type-tools/afdko/issues/470, and this is how I rewrote `__repr__`:

```python
    def __repr__(self):
        fddict = {}
        for key, val in vars(self).items():
            if val is None:
                continue
            fddict[key] = val
        return "<%s '%s' %s>" % (
            self.__class__.__name__, fddict.get('DictName', 'no name'), fddict)
```

Reply:
It is a stripped down version of that file indeed. BTW, I have been thinking, since  AFDKO is on PyPI now, should we drop all three copied and stripped down modules and import AFDKO’s directly (and possibly upstreaming any fixes I did)? Or is it too early for that?

Reply:
no, if anything, `afdko` should depend on `psautohint`, not the other way around.

I plan to make `psautohint` an `afdko` dependency sometime soon. Once we're confident that `autohint` can be retired, I'll look into stripping the `afdko` parts that are no longer needed.